contracts/RoyaltyFeeSetterV2.sol

```diff
@@ -0,0 +1,191 @@
+error RoyaltyFeeSetterV2__NotCollectionSetter();
+error RoyaltyFeeSetterV2__SetterAlreadySet();
+
+/**
+ * @title RoyaltyFeeSetter
+ * @notice Used to allow creators to set royalty information in RoyaltyFeeRegistryV2.
+ */
+contract RoyaltyFeeSetterV2 is Initializable, OwnableUpgradeable {
+    using RoyaltyFeeTypes for RoyaltyFeeTypes.FeeInfoPart;
+
+    // ERC721 interfaceID
+    bytes4 public constant INTERFACE_ID_ERC721 = 0x80ac58cd;
+
+    // ERC1155 interfaceID
+    bytes4 public constant INTERFACE_ID_ERC1155 = 0xd9b67a26;
+
+    // ERC2981 interfaceID
+    bytes4 public constant INTERFACE_ID_ERC2981 = 0x2a55205a;
+
+    IRoyaltyFeeRegistryV2 public royaltyFeeRegistryV2;
+
+    /**
+     * @notice Initializer
+     * @param _royaltyFeeRegistryV2 address of the royalty fee registry
+     */
+    function initialize(address _royaltyFeeRegistryV2) public initializer {
+        __Ownable_init();
+
+        royaltyFeeRegistryV2 = IRoyaltyFeeRegistryV2(_royaltyFeeRegistryV2);
+    }
+
+    /**
+     * @notice Update royalty info for collection if admin
+     * @dev Only to be called if there is no setter address
+     * @param _collection address of the NFT contract
+     * @param _setter address that sets the receiver
+     * @param _feeInfoParts fee info parts
+     */
+    function updateRoyaltyInfoPartsForCollectionIfAdmin(
+        address _collection,
+        address _setter,
+        RoyaltyFeeTypes.FeeInfoPart[] memory _feeInfoParts
+    ) external {
+        if (IERC165(_collection).supportsInterface(INTERFACE_ID_ERC2981)) {
+            revert RoyaltyFeeSetterV2__CollectionCannotSupportERC2981();
+        }
+        if (msg.sender != IOwnable(_collection).admin()) {
+            revert RoyaltyFeeSetterV2__NotCollectionAdmin();
+        }
+        _updateRoyaltyInfoPartsForCollectionIfOwnerOrAdmin(
+            _collection,
+            _setter,
+            _feeInfoParts
+        );
+    }
+
+    /**
+     * @notice Update royalty info for collection if owner
+     * @dev Only to be called if there is no setter address
+     * @param _collection address of the NFT contract
+     * @param _setter address that sets the receiver
+     * @param _feeInfoParts fee info parts
+     */
+    function updateRoyaltyInfoPartsForCollectionIfOwner(
+        address _collection,
+        address _setter,
+        RoyaltyFeeTypes.FeeInfoPart[] memory _feeInfoParts
+    ) external {
+        if (IERC165(_collection).supportsInterface(INTERFACE_ID_ERC2981)) {
+            revert RoyaltyFeeSetterV2__CollectionCannotSupportERC2981();
+        }
+        if (msg.sender != IOwnable(_collection).owner()) {
+            revert RoyaltyFeeSetterV2__NotCollectionOwner();
+        }
+        _updateRoyaltyInfoPartsForCollectionIfOwnerOrAdmin(
+            _collection,
+            _setter,
+            _feeInfoParts
+        );
+    }
+
+    /**
+     * @notice Update royalty info for collection
+     * @dev Only to be called if there msg.sender is the setter
```

Comment:
`@dev Only to be called if there msg.sender is the setter` typo `if *the* msg.sender`